doc/design/block.md

```diff
@@ -73,48 +76,59 @@ An equivalent PaddlePaddle program from the design doc of the [IfElseOp operator
 ```python
 import paddle as pd
 
-x = var(10)
-y = var(20)
+x = var([10, 20])
+# a scalar
+y = var([1])
+z = var(10, 20)
 cond = var(false)
-ie = pd.create_ifelseop(inputs=[x], output_num=1)
+# output_num should be set to ensure the outputs of the true_block and false_block can be merged,
+# so the numbers of both blocks should be same as `output_num`.
+ie = pd.ifelse_builder(output_num=2)
+
 with ie.true_block():
-    x = ie.inputs(true, 0)
-    z = operator.add(x, y)
-    ie.set_output(true, 0, operator.softmax(z))
+    x_ = x.as_ifelse_input()
```

Comment:
If the framework knows whether x contains instances, user does not indicate x  as ifelse_input. The use can directly use x in the block and the framework can automatically do the splitting.

Reply:
some x that has instances might not need to be split.

such as
```python
# v is some op's output
v = some_op() # shape is [20, 20]
data = var(shape=[20, 20])

with ie.true_block():
    # split x
    x = data.as_ifelse_input() # shape [1, 20]
    # v should not be split
    y = pd.matmul(x.T, v) # shape [1, 20]
   y_T = y.T # [20, 1]
   ie.set_outputs(y_T)
```
v has the same batch_size, but do not need to be split.

if write as 

```
with ie.true_block():
    y = pd.matmul(x, v) # shape [1, 20] x [1, 20] wrong
```
the shapes will not match.

Reply:
If a variable has fixed size, it's not splittable. If a variable's size depends on batchsize, it must be splitted because it means that it contains data for each instances.

Reply:
oh, I see, I will change this latter.